sentinel-adapter/sentinel-sofa-rpc-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/sofa/rpc/SentinelSofaRpcProviderFilter.java

```diff
@@ -0,0 +1,116 @@
+import com.alipay.sofa.rpc.core.exception.RpcErrorType;
+import com.alipay.sofa.rpc.core.exception.SofaRpcException;
+import com.alipay.sofa.rpc.core.request.SofaRequest;
+import com.alipay.sofa.rpc.core.response.SofaResponse;
+import com.alipay.sofa.rpc.ext.Extension;
+import com.alipay.sofa.rpc.filter.AutoActive;
+import com.alipay.sofa.rpc.filter.FilterInvoker;
+
+import static com.alibaba.csp.sentinel.adapter.sofa.rpc.SofaRpcUtils.getApplicationName;
+import static com.alibaba.csp.sentinel.adapter.sofa.rpc.SofaRpcUtils.getInterfaceResourceName;
+import static com.alibaba.csp.sentinel.adapter.sofa.rpc.SofaRpcUtils.getMethodResourceName;
+import static com.alibaba.csp.sentinel.adapter.sofa.rpc.SofaRpcUtils.getMethodArguments;
+
+/**
+ * SOFARPC service provider filter for Sentinel, auto activated by default.
+ *
+ * If you want to disable the provider filter, you can configure:
+ * <pre>ProviderConfig.setParameter("sofa.rpc.sentinel.enabled", "false");</pre>
+ *
+ * or add setting in rpc-config.json file:
+ * <pre>
+ * {
+ *   "sofa.rpc.sentinel.enabled": false
+ * }
+ * </pre>
+ *
+ * @author cdfive
+ */
+@Extension(value = "providerSentinel", order = -1000)
+@AutoActive(providerSide = true)
+public class SentinelSofaRpcProviderFilter extends AbstractSofaRpcFilter {
+
+    @Override
+    public SofaResponse invoke(FilterInvoker invoker, SofaRequest request) throws SofaRpcException {
+        // Now only support sync invoke.
+        if (request.getInvokeType() != null && !RpcConstants.INVOKER_TYPE_SYNC.equals(request.getInvokeType())) {
+            return invoker.invoke(request);
+        }
+
+        String applicationName = getApplicationName(request);
+        String interfaceResourceName = getInterfaceResourceName(request);
+        String methodResourceName = getMethodResourceName(request);
+
+        Entry interfaceEntry = null;
+        Entry methodEntry = null;
+        try {
+            ContextUtil.enter(methodResourceName, applicationName);
+
+            interfaceEntry = SphU.entry(interfaceResourceName, ResourceTypeConstants.COMMON_RPC, EntryType.IN);
+            methodEntry = SphU.entry(methodResourceName, ResourceTypeConstants.COMMON_RPC, EntryType.IN, getMethodArguments(request));
+
+            SofaResponse response = invoker.invoke(request);
+
+            response.setErrorMsg("test error msg");
+
+            if (response.isError()) {
+                SofaRpcException rpcException = new SofaRpcException(RpcErrorType.SERVER_FILTER, response.getErrorMsg());
+                Tracer.traceEntry(rpcException, interfaceEntry);
+                Tracer.traceEntry(rpcException, methodEntry);
+            } else {
+                Object appResponse = response.getAppResponse();
+                if (appResponse instanceof Throwable) {
+                    Tracer.traceEntry((Throwable) appResponse, interfaceEntry);
+                    Tracer.traceEntry((Throwable) appResponse, methodEntry);
+                }
+            }
```

Comment:
Maybe this code snippet can be extracted into `AbstractSofaRpcFilter`?

Reply:
Good idea! Two method, one for trace response exception, and one for trace other exception has been refactored into `AbstractSofaRpcFilter`.